libgalois/src/analytics/louvain_clustering/louvain_clustering.cpp

```diff
@@ -173,6 +173,200 @@ struct LouvainClusteringImplementation
     return prev_mod;
   }
 
+  katana::Result<double> LouvainDeterministic(
```

Comment:
This function looks very similar to the one above. Can we abstract things a bit and reduce the duplication? The simplest approach would be to lift some common elements into a shared function or three. But it might also be possible to combine the functions entirely and just have a flag parameter to switch modes. (or template parameter if we must for some reason.)

Reply:
I am not sure of how to do that at the moment. I will think more on this.

Reply:
For the moment, you can just document it with a `TODO` and move on. I'm not sure how to pull out the duplication either. Or maybe there isn't as much duplication as it appears. I may not be understanding.